modules/storages/app/components/storages/admin/storage_list_component.html.erb

```diff
@@ -0,0 +1,78 @@
+<%=
+  if storages.present?
+    render(Primer::Beta::BorderBox.new) do |component|
+      component.with_header(color: :subtle) do
+        grid_layout('op-storage-list', tag: :div, align_items: :center) do |grid|
+          grid.with_area(:name, tag: :div) do
+            render(Primer::Beta::Text.new(font_weight: :semibold)) { I18n.t('storages.label_name') }
+          end
+
+          grid.with_area(:user, tag: :div, color: :subtle, mr: 3) do
+            render(Primer::Beta::Text.new(font_weight: :semibold)) { I18n.t('storages.label_creator') }
+          end
+
+          grid.with_area(:provider, tag: :div, color: :subtle, mr: 3) do
+            render(Primer::Beta::Text.new(font_weight: :semibold)) { I18n.t('storages.label_provider') }
+          end
+
+          grid.with_area(:time, tag: :div, color: :subtle) do
+            render(Primer::Beta::Text.new(font_weight: :semibold)) { I18n.t('storages.label_creation_time') }
+          end
+        end
+      end
+
+      storages.map do |storage|
+        component.with_row(scheme: :default, id: storage_row_css_id(storage)) do
+          grid_layout('op-storage-list', tag: :div, align_items: :center) do |grid|
+            grid.with_area(:name, tag: :div, mr: 1) do
+              render(Primer::Beta::Link.new(href: edit_admin_settings_storage_path(storage), scheme: :primary)) { storage.name }
+            end
+
+            unless storage.configured?
+              grid.with_area(:label, tag: :div) do
+                render(Primer::Beta::Label.new(scheme: :attention)) { I18n.t('storages.label_incomplete') }
+              end
+            end
+
+            grid.with_area(:host, tag: :div, color: :subtle) do
+              render(Primer::Beta::Truncate.new(font_weight: :light)) { storage.host }
+            end
+
+            grid.with_area(:user, tag: :div, color: :subtle, mr: 3) do
+              render(Users::AvatarComponent.new(user: storage.creator, size: :mini, link: false, show_name: true))
+            end
+
+            grid.with_area(:provider, tag: :div, color: :subtle, mr: 3) do
+              render(Primer::Beta::Truncate.new(font_weight: :light)) { storage.short_provider_type.capitalize }
+            end
+
+            grid.with_area(:time, tag: :div, color: :subtle, mr: 3) do
+              render(Primer::Beta::Text.new(font_weight: :light)) { storage.created_at.strftime(I18n.t('datetime.formats.long')) }
+            end
+          end
+        end
+      end
+    end
+  else
+    render Primer::Beta::Blankslate.new(border: true) do |component|
+      component.with_visual_icon(icon: :cloud)
+      component.with_heading(tag: :h2).with_content(I18n.t('storages.storage_list_blank_slate.heading'))
+      component.with_description { I18n.t('storages.storage_list_blank_slate.description') }
+
+      component.with_primary_action(
+        href: new_admin_settings_storage_path,
```

Comment:
They are, although It seemed rather hacky to try and reuse them as primer primary_action slot is tightly coupled- Ideally it would have been better to just pass the `NewStorageButtonComponent`.

We could however do sth like

```ruby
component.with_primar_action(**NewStorageButtonComponent.button_options) do
  button.with_leading_visual_icon(icon: :plus)
  I18n.t("storages.label_storage")
end

```